confluent_platform/tests/metrics.py

```diff
@@ -0,0 +1,269 @@
+    'kafka.producer.io_ratio',
+    'kafka.producer.io_wait_ratio',
+    'kafka.producer.outgoing_byte_rate',
+    'kafka.producer.produce_throttle_time_avg',
+    'kafka.producer.produce_throttle_time_max',
+    'kafka.producer.record_error_rate',
+    'kafka.producer.record_retry_rate',
+    'kafka.producer.waiting_threads',
+    'kafka.producer.connection_count',
+    'kafka.producer.network_io_rate',
+    'kafka.producer.request_rate',
+    'kafka.producer.response_rate',
+    'kafka.producer.request_latency_avg',
+    'kafka.producer.io_wait_time_ns_avg',
+    'kafka.producer.connection_close_rate',
+    'kafka.producer.connection_creation_rate',
+    'kafka.producer.io_time_ns_avg',
+    'kafka.producer.select_rate',
+    'kafka.producer.incoming_byte_rate',
+]
+
+PRODUCER_NODE_METRICS = [
+    'kafka.producer.node.incoming_byte_rate',
+    'kafka.producer.node.outgoing_byte_rate',
+    'kafka.producer.node.request_rate',
+    'kafka.producer.node.request_size_avg',
+    'kafka.producer.node.request_size_max',
+    'kafka.producer.node.response_rate',
+]
+
+PRODUCER_TOPIC_METRICS = [
+    'kafka.producer.topic.byte_rate',
+    'kafka.producer.topic.compression_rate',
+    'kafka.producer.topic.record_error_rate',
+    'kafka.producer.topic.record_retry_rate',
+    'kafka.producer.topic.record_send_rate',
+]
+
+CONSUMER_METRICS = [
+    'kafka.consumer.io_ratio',
+    'kafka.consumer.io_wait_ratio',
+    'kafka.consumer.connection_count',
+    'kafka.consumer.network_io_rate',
+    'kafka.consumer.request_rate',
+    'kafka.consumer.response_rate',
+]
+
+CONSUMER_NODE_METRICS = []
+
+
+CONSUMER_FETCH_METRICS = [
+    'kafka.consumer.fetch.bytes_consumed_rate',
+    'kafka.consumer.fetch.fetch_latency_avg',
+    'kafka.consumer.fetch.fetch_latency_max',
+    'kafka.consumer.fetch.fetch_rate',
+    'kafka.consumer.fetch.fetch_throttle_time_avg',
+    'kafka.consumer.fetch.fetch_throttle_time_max',
+    'kafka.consumer.fetch.records_consumed_rate',
+]
+
+CONSUMER_FETCH_TOPIC_METRICS = [
+    'kafka.consumer.fetch_topic.bytes_consumed_rate',
+    'kafka.consumer.fetch_topic.records_consumed_rate',
+]
+
+KSQL_QUERY_STATS = [
+    'confluent.ksql.query_stats.bytes_consumed_total',
+    'confluent.ksql.query_stats.error_rate',
+    'confluent.ksql.query_stats.messages_consumed_per_sec',
+    'confluent.ksql.query_stats.messages_consumed_total',
+    'confluent.ksql.query_stats.messages_produced_per_sec',
+    'confluent.ksql.query_stats.num_active_queries',
+    'confluent.ksql.query_stats.num_idle_queries',
+    'confluent.ksql.query_stats.num_persistent_queries',
+]
+
+ALWAYS_PRESENT_METRICS = (
+    BROKER_METRICS
+    + CONNECT_METRICS
+    + REST_JETTY_METRICS
+    + SCHEMA_REGISTRY_JETTY_METRICS
+    + SCHEMA_REGISTRY_METRICS
+    + PRODUCER_METRICS
+    + PRODUCER_NODE_METRICS
+    + PRODUCER_TOPIC_METRICS
+    + CONSUMER_METRICS
+    + CONSUMER_NODE_METRICS
+    + KSQL_QUERY_STATS
+)
+
+NOT_ALWAYS_PRESENT_METRICS = (
```

Comment:
Can we get a comment here to explain why those are not always present? Test environment related or confluent related?

Reply:
Added a comment about that, let me know if that's ok.